### PR TITLE
feat(!): converts to typescript

### PR DESCRIPTION
**Pull request by achingbrain**
- Only named exports
- No more CJS, only ESM
- Adds auto-publish
- Adds dependabot

BREAKING CHANGE: switch to named exports, ESM only

### CODE REVIEW

**Comment by achingbrain** [comment]
N.b. to get the auto-publish goodness, create a repository secret here called `NPM_TOKEN` with a npm automation token that can publish to `is-loopback-addr`.

File: .github/workflows/js-test-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+name: test & maybe release
+on:
+  push:
+    branches:
+      - master
```
Comment (vasco-santos) [inline .github/workflows/js-test-and-release.yml]:
```suggestion
      - main
```

File: .github/workflows/js-test-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  pull_request:
+    branches:
+      - master
```
Comment (vasco-santos) [inline .github/workflows/js-test-and-release.yml]:
```suggestion
      - main
```

**Review by vasco-santos — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,12 +1,11 @@
 # is-loopback-addr
 
-[![Build Status](https://travis-ci.org/vasco-santos/is-loopback-addr.svg?branch=main)](https://travis-ci.org/vasco-santos/is-loopback-addr)
+[![Build Status](https://github.com/vasco-santos/is-loopback-addr/actions/workflows/main.yml/badge.svg?branch=master)](https://github.com/vasco-santos/is-loopback-addr/actions/workflows/main.yml)
```
Comment (vasco-santos) [inline README.md]:
```suggestion
[![Build Status](https://github.com/vasco-santos/is-loopback-addr/actions/workflows/js-test-and-release.yml/badge.svg?branch=main)](https://github.com/vasco-santos/is-loopback-addr/actions/workflows/js-test-and-release.yml)
```

File: package.json [hunk 4]
```diff
@@ -2,27 +2,131 @@
...
+  },
+  "eslintConfig": {
+    "extends": "ipfs",
+    "parserOptions": {
+      "sourceType": "module"
+    }
+  },
+  "release": {
+    "branches": [
+      "master"
```
Comment (vasco-santos) [inline package.json]:
```suggestion
      "main"
```

File: .github/workflows/js-test-and-release.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - run: npx xvfb-maybe npm run --if-present test:electron-renderer
+
+  release:
+    needs: [test-node, test-chrome, test-chrome-webworker, test-firefox, test-firefox-webworker, test-electron-main, test-electron-renderer]
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```
Comment (vasco-santos) [inline .github/workflows/js-test-and-release.yml]:
```suggestion
    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

**Review by vasco-santos — CHANGES_REQUESTED** [review]

**Review by vasco-santos — APPROVED** [review]
LGTM thanks ❤️

**Pull request merged**